chain-impl-mockchain/src/vote/payload.rs

```diff
@@ -18,11 +20,26 @@ use typed_bytes::ByteBuilder;
 #[repr(u8)]
 pub enum PayloadType {
     Public = 1,
+    Private = 2,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum Payload {
     Public { choice: Choice },
+    Private { encrypted_vote: EncryptedVote },
```

Comment:
Question: don't we need the `ProofOfCorrectVote` here too? Or how do we distribute those?

Reply:
Im not sure. I think the proof is for the voter to check the encrypted onchain vote. @rinor what do you think?

Reply:
IMO we should have it here so that everybody could validate the vote, not only the node that included this vote into a block.

Reply:
Furthermore the `Proof` can be used from the node receiving the VoteCast transaction to check if indeed the correct `ElectionPublicKey` (EncryptingVoteKey) was used and reject the transaction if not.

Reply:
I don't know the details, but I though you needed the proof to validate that someone didn't vote for two options, or that the vector is actually unitary. That seems like something that each participant in the network should be able to validate.

Reply:
Yes, we should do that indeed. Where should I include that check? Can @eugene-babichenko or @enzoc4 point into some snippet please?

Reply:
Probably manager is the right place https://github.com/input-output-hk/chain-libs/blob/master/chain-impl-mockchain/src/vote/manager.rs#L106